scipy/_lib/_array_api.py

```diff
@@ -196,19 +196,26 @@ def is_torch(xp):
     return xp.__name__ == 'scipy._lib.array_api_compat.array_api_compat.torch'
 
 
-def _strict_check(actual, desired, check_xp=True, check_dtype=True, check_shape=True):
-    if check_xp:
+def _strict_check(actual, desired, xp,
+                  check_namespace=True, check_dtype=True, check_shape=True):
+    if check_namespace:
         _assert_matching_namespace(actual, desired)
-    xp = array_namespace(actual)
+
     actual = xp.asarray(actual)
     desired = xp.asarray(desired)
+
     if check_dtype:
         assert_(actual.dtype == desired.dtype,
-                f"Desired dtype: {desired.dtype}, actual dtype: {actual.dtype}")
+                "dtypes do not match.\n"
```

Comment:
The particular format doesn't matter, but we might as well pick a *consistent* format for all of these messages.